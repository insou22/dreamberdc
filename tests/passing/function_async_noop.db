async func count() {
   print(2)
   noop!
   print(5)!
}

count()!
print(1)!
print(3)!
print(4)!