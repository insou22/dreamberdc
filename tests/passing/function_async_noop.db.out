Program {
    items: [
        FnDecl(
            FnDecl {
                is_async: true,
                keyword: "func",
                name: Ident {
                    name: "count",
                },
                args: [],
                body: BlockBody(
                    [
                        Expr(
                            ExpressionStatement {
                                expr: FreeFnCall(
                                    FreeFnCallExpr {
                                        fn_name: Ident {
                                            name: "print",
                                        },
                                        params: [
                                            NumericLit(
                                                NumericLit {
                                                    digits: "2",
                                                },
                                            ),
                                        ],
                                    },
                                ),
                                debug: false,
                            },
                        ),
                        Noop(
                            NoopStatement,
                        ),
                        Expr(
                            ExpressionStatement {
                                expr: FreeFnCall(
                                    FreeFnCallExpr {
                                        fn_name: Ident {
                                            name: "print",
                                        },
                                        params: [
                                            NumericLit(
                                                NumericLit {
                                                    digits: "5",
                                                },
                                            ),
                                        ],
                                    },
                                ),
                                debug: false,
                            },
                        ),
                    ],
                ),
            },
        ),
        Statement(
            Expr(
                ExpressionStatement {
                    expr: FreeFnCall(
                        FreeFnCallExpr {
                            fn_name: Ident {
                                name: "count",
                            },
                            params: [],
                        },
                    ),
                    debug: false,
                },
            ),
        ),
        Statement(
            Expr(
                ExpressionStatement {
                    expr: FreeFnCall(
                        FreeFnCallExpr {
                            fn_name: Ident {
                                name: "print",
                            },
                            params: [
                                NumericLit(
                                    NumericLit {
                                        digits: "1",
                                    },
                                ),
                            ],
                        },
                    ),
                    debug: false,
                },
            ),
        ),
        Statement(
            Expr(
                ExpressionStatement {
                    expr: FreeFnCall(
                        FreeFnCallExpr {
                            fn_name: Ident {
                                name: "print",
                            },
                            params: [
                                NumericLit(
                                    NumericLit {
                                        digits: "3",
                                    },
                                ),
                            ],
                        },
                    ),
                    debug: false,
                },
            ),
        ),
        Statement(
            Expr(
                ExpressionStatement {
                    expr: FreeFnCall(
                        FreeFnCallExpr {
                            fn_name: Ident {
                                name: "print",
                            },
                            params: [
                                NumericLit(
                                    NumericLit {
                                        digits: "4",
                                    },
                                ),
                            ],
                        },
                    ),
                    debug: false,
                },
            ),
        ),
    ],
}